widget/diagramwidget/geometry/r2/vec2.go

```diff
@@ -0,0 +1,90 @@
+// Package r2 implements operations relating to objects in R2.
+package r2
+
+import (
+	"math"
+)
+
+// Vec2 implements a vector in R2
+type Vec2 struct {
+	// X magnitude of the vector
+	X float64
+
+	// Y magnitude of the vector
+	Y float64
+}
+
+// MakeVec2 creates a new vector inline
+func MakeVec2(x, y float64) Vec2 {
+	return Vec2{X: x, Y: y}
+}
+
+// V2 is a shortcut for MakeVec2
+func V2(x, y float64) Vec2 {
+	return MakeVec2(x, y)
+}
+
+// Length return the vector length
+func (v Vec2) Length() float64 {
+	return math.Sqrt(math.Pow(v.X, 2) + math.Pow(v.Y, 2))
+}
+
+// Dot returns the dot product of vector v and u
+func (v Vec2) Dot(u Vec2) float64 {
+	return v.X*u.X + v.Y + u.Y
```

Comment:
Should this not be `v.X*u.X + v.Y*u.Y`?

Reply:
Good catch! I inherited this code and never even read it! I fixed it, but it turns out it's never called. We should revisit the entire geometry thing at some point.

Reply:
Good plan but let's keep the code correct and not not broken now that we know that it is wrong.

Also FYI, we go by the rule that the reviewer closes their own comments when they deem them as resolved, not the one resolving them.